File: common/src/main/java/ArenaInfo.java
import com.fasterxml.jackson.annotation.JsonProperty;

import java.security.SecureRandom;


record ArenaInfo(
        @JsonProperty("defendZoneTopLeft") Point defendZoneTopLeft,
        @JsonProperty("defendZoneBottomRight") Point defendZoneBottomRight,
        @JsonProperty("goal") Point goal
) implements Jsonisable {
    public Point randomDefendPoint(SecureRandom rnd) {
        var diag = defendZoneBottomRight.diff(defendZoneTopLeft);
        return new Point(rnd.nextInt((int) diag.x()), rnd.nextInt((int) diag.y()))
                .add(defendZoneTopLeft);
    }
}

File: common/src/main/kotlin/demo/mesharena/common/ArenaInfo.kt
package demo.mesharena.common

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import java.security.SecureRandom

data class ArenaInfo @JsonCreator constructor(
        @JsonProperty("defendZoneTopLeft") val defendZoneTopLeft: Point,
        @JsonProperty("defendZoneBottomRight") val defendZoneBottomRight: Point,
        @JsonProperty("goal") val goal: Point
) : Jsonisable {
    fun randomDefendPoint(rnd: SecureRandom): Point {
        val (x, y) = defendZoneBottomRight.diff(defendZoneTopLeft)
        return Point(rnd.nextInt(x.toInt()).toDouble(), rnd.nextInt(y.toInt()).toDouble())
                .add(defendZoneTopLeft)
    }
}